### PR TITLE
Fix UI rendering issues if XCom values are INT, FLOAT, BOOL or NULL in web UI

### LINKED ISSUE #41514: XCom Display for INT value shows error in web UI
### Apache Airflow version

2.10.0rc1

### If "Other Airflow 2 version" selected, which one?

_No response_

### What happened?

When one task returns an integer value, the "new" XCom display shows an error JSON instead of the value like:

image

Noteworthy>: JSON or string or empty value works.

### What you think should happen instead?

The real INT value should be displayed

### How to reproduce

Run the DAG example_dynamic_task_mapping and check the XCom result of "add_one" task

### Operating System

not relevant

### Versions of Apache Airflow Providers

not relevant

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

not relevant

### Anything else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by jscheffl**
closes: #41514 

Before:
image

After:
image

### CODE REVIEW

**Review by kaxil — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

File: airflow/www/static/js/components/RenderedJsonField.tsx [hunk 1]
```diff
@@ -47,7 +47,7 @@ const JsonParse = (content: string | object) => {
       contentJson = content;
     }
     contentFormatted = JSON.stringify(contentJson, null, 4);
-    isJson = true;
+    isJson = Number.isNaN(contentJson); // ensure numbers/bool are not treated as JSON
```
Comment (uranusjr) [inline airflow/www/static/js/components/RenderedJsonField.tsx]: I think this actually returns `false` for everything that’s actually JSON…

```js
var c = JSON.stringify({a: 1})
console.log(Number.isNaN(c))  // false
```

**Review by uranusjr — CHANGES_REQUESTED** [review]

File: airflow/www/static/js/components/RenderedJsonField.tsx (hunk 1 above)
Comment (jscheffl) [reply airflow/www/static/js/components/RenderedJsonField.tsx]: Uuups. Yes. Reading docs helps. Still I am wondering why actually it _was_ working with this.

Anyway, how about as I pushed a different approach? After parsing as JSON should be an object (which is true for Array+Dict)

File: airflow/www/static/js/components/RenderedJsonField.tsx [hunk 2]
```diff
@@ -47,7 +47,7 @@ const JsonParse = (content: string | object) => {
       contentJson = content;
     }
     contentFormatted = JSON.stringify(contentJson, null, 4);
-    isJson = true;
+    isJson = typeof contentJson === "object"; // ensure numbers/bool are not treated as JSON
```
Comment (uranusjr) [inline airflow/www/static/js/components/RenderedJsonField.tsx]: This also excludes `null`. I’m not even sure if it’s a possible value here, or how we are expected to treat it.

Comment (jscheffl) [reply airflow/www/static/js/components/RenderedJsonField.tsx]: Oh. You are really good. Did not have "NULL" in my test set as in all my examples NULL was never persisted in XCom, in these cases just no XCom was stored. But actually if I try we have even a second problem with NULL in the XCom value. API returns an error so... need to fix this as well.

Added NULL to the party, updated screenshot in PR above how it comes out

Comment (jscheffl) [reply airflow/www/static/js/components/RenderedJsonField.tsx]: Okay, got a bit more complex now but extracted the utility function and added a test for it.

**Review by bbovenzi — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

**Review by uranusjr — APPROVED** [review]

**Pull request merged**